### DISCUSSION TITLE
distributed deployment of qdrant

### DISCUSSION THREAD

**Discussion by mm3906078** [qdrant/qdrant#4341] [Q&A] [answered]
Hi folks, if we have multiple instances of quadrant, one marked as a leader, and the other should be a flower, how does the quadrant handle write from flowers? What happens if data should be written in shards that are not in the local flower data?

**Comment by timvisee** [comment, marked as answer]
Leader election happens automatically. Any node can be the leader. The elected leader might change if a node goes down.

Data is spread across nodes in (replicated) shards. If a shard doesn't have a local replica, any other node having a replica is contacted and used.

Please read through our distributed deployment documentation here, it elaborates on all of this in detail: https://qdrant.tech/documentation/guides/distributed_deployment/

By default, all nodes accept reads and writes and updates are not serialized through the leader. If you'd like to have strong consistency guarantees you can configure it as described here: https://qdrant.tech/documentation/guides/distributed_deployment/#consistency-guarantees
